index.d.ts

```diff
@@ -0,0 +1,6301 @@
+         */
+        selectable?: boolean | undefined;
+        /**
+         * When locked a node's position is immutable (default false)
+         */
+        locked?: boolean | undefined;
+        /**
+         * Wether the node can be grabbed and moved by the user
+         */
+        grabbable?: boolean | undefined;
+        /**
+         * Whether the element has passthrough panning enabled.
+         */
+        pannable?: boolean | undefined;
+        /**
+         * a space separated list of class names that the element has
+         */
+        classes?: string[] | string | undefined;
+        /**
+         *  CssStyleDeclaration;
+         */
+        style?: CssStyleDeclaration | undefined;
+        /**
+         * you should only use `style`/`css` for very special cases; use classes instead
+         */
+        css?: Css.Node | Css.Edge | undefined;
+    }
+
+    interface ElementDataDefinition {
+        /**
+         * elided id => autogenerated id
+         */
+        id?: string | undefined;
+        position?: Position | undefined;
+    }
+
+    interface EdgeDefinition extends ElementDefinition {
+        data: EdgeDataDefinition;
+    }
+
+    interface EdgeDataDefinition extends ElementDataDefinition {
+        id?: string | undefined;
+
+        /**
+         * the source node id (edge comes from this node)
+         */
+        source: string;
+        /**
+         * the target node id (edge goes to this node)
+         */
+        target: string;
+
+        [key: string]: any;
+    }
+
+    interface NodeDefinition extends ElementDefinition {
+        data: NodeDataDefinition;
+    }
+
+    interface NodeDataDefinition extends ElementDataDefinition {
+        id?: string | undefined;
+        parent?: string | undefined;
+        [key: string]: any;
+    }
+
+    interface CytoscapeOptions {
+        ///////////////////////////////////////
+        // very commonly used options:
+        /**
+         * A HTML DOM element in which the graph should be rendered.
+         * This is optional if Cytoscape.js is run headlessly or if you initialise using jQuery (in which case your jQuery object already has an associated DOM element).
+         *
+         * The default is undefined.
+         */
+        container?: HTMLElement | null | undefined;
+
+        /**
+         * An array of [[Elements]] specified as plain objects. For convenience, this option can alternatively be specified as a promise that resolves to the elements JSON.
+         */
+        elements?:
+            | ElementsDefinition
+            | ElementDefinition[]
+            | Promise<ElementsDefinition>
+            | Promise<ElementDefinition[]>
+            | undefined;
+        /**
+         * The [[Stylesheet]] used to style the graph. For convenience, this option can alternatively be specified as a promise that resolves to the stylesheet.
+         */
+        style?: Stylesheet[] | Promise<Stylesheet[]> | undefined;
```

Comment:
The style property is an array of StyleSheets? This naming seems a bit strange.

For example:
```javascript
style: [
  { selector: "node", 
    style: { 
      width: 20, 
      height: 20 
    } 
  },
  { selector: "edge", 
    style: { 
      "curve-style": "unbundled-bezier", 
      "control-point-weight": 0.5, 
      "control-point-distance": 20 
    } 
  },
],
```

According to this type definition each object in the array has type Stylesheet. Is that right? Don't we think of the entire array as a Stylesheet? With each element being a style definition or style block.